pyblish/util.py

```diff
@@ -26,13 +26,62 @@ def publish(context=None, plugins=None, targets=None):
             defaults to results of discover()
         targets (list, optional): Targets to include for publish session.
 
+    Returns:
+        Context: The context processed by the plugins.
+
     Usage:
         >> context = plugin.Context()
         >> publish(context)     # Pass..
         >> context = publish()  # ..or receive a new
 
     """
 
+    # Need to generate a Context object if None is passed. This is because,
```

Comment:
I don't understand this? Can you give me an example?

Reply:
If we pass in ```None``` as the context, ei. don't pass in any context to the method, the ```publish_iter``` method won't process the context object, and when we return the context object (https://github.com/pyblish/pyblish-base/pull/326/files/1306986e78f49655c131eb1787dd76d31bf4f716#diff-45fca2405e95038f14f9537f62615994R49), then it'll be ```None``` from the default value.
More specifically the two yields in ```publish_iter``` don't get executed, so the context object does not get modified. This could be resolved by always yielding the context at the end of ```publish_iter```, but then we get into the issue of yielding different results per iteration, which in my opinion makes it tricky to work with.

Reply:
Aah, ok, now I understand. Yes that makes sense, but the description could use some work. In fact, I don't think you need it, it's pretty clear you need it in order to return it.

Reply:
Fair enough, I can get rid of the comment :)